### PR TITLE
cdc: optionally output old value

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

* Define `ExtraOp`
* Add `ExtraOp` option to `ChangeDataRequest`
* Add `old_value` to `Event`

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -858,3 +858,9 @@ message KeyRange {
     bytes start_key = 1;
     bytes end_key = 2;
 }
+
+enum ExtraRead {
+    Noop = 0;
+    Deleted = 1;
+    Updated = 2;
```
Comment (amyangfei) [inline proto/kvrpcpb.proto:865]: Does `Deleted` mean only `delete` event contains the previous value, and `Updated` mean `put` event (insert or update) contains the previous value?
Maybe we need a mode that both `delete` and `put` contain the previous value.

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -858,3 +858,9 @@ message KeyRange {
     bytes start_key = 1;
     bytes end_key = 2;
 }
+
+enum ExtraRead {
```
Comment (MyonKeminta) [inline proto/kvrpcpb.proto:862]: Why is this defined in `kvrpcpb.proto`?

Comment (5kbpers) [reply proto/kvrpcpb.proto:862]: Both `storage` and `raftstore` will use it.

Comment (overvenus) [reply proto/kvrpcpb.proto]: It's okay to put them in cdcpb as cdcpb does not depend on both `raftstore` and `storage`.

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -861,3 +861,9 @@ message KeyRange {
     bytes start_key = 1;
     bytes end_key = 2;
 }
+
+enum ExtraOp {
+    Noop = 0;
+    ReadDeleted = 1;
+    ReadOldValue = 2;
```
Comment (overvenus) [inline proto/kvrpcpb.proto:868]: Can we have them both set in the future?

Comment (5kbpers) [reply proto/kvrpcpb.proto:868]: `ReadDeleted` is included in `ReadOldValue`.

Comment (overvenus) [reply proto/kvrpcpb.proto:868]: Could you add some comments? The behavior is not obvious to me.

File: proto/kvrpcpb.proto (hunk 2 above)
Comment (5kbpers) [reply proto/kvrpcpb.proto]: But it's a switch which could control the operation in `txn`, so I guess it's a little strange to put it to cdcpb.🤔

**Pull request merged**